### DISCUSSION TITLE
query meta @attributes from .cue script

### DISCUSSION THREAD

**Discussion by proppy** [cuelang/cue#396] [General]
I'd like to be able to leverage annotations within `cue cmd` tooling to create tasks programmatically, is it possible to query `@attributes` annotations from within CUE?

**Comment by verdverm** [comment]
I don't believe so, there is also resistance to making this possible within Cue directly. 

- The spec says attributes "are for non-cue things" (more technically). They are not part of the unification process, so cannot be used within Cue code and the evaluator engine. 
- The tooling layer may be acceptable (@mpvl ?) but not likely.
- The recommended way is to use the Go API. (you can see an example here: https://github.com/hofstadter-io/hof/blob/master/lib/structural/find_attrs.go)

That being said, there is one attribute `@tag(...)` which can be used from the CLI with the `-t/--inject` flag which works from several sub commands and in pure Cue code.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/396.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
